crucible-syntax/src/Lang/Crucible/Syntax/Concrete.hs

```diff
@@ -680,14 +793,16 @@ synthExpr typeHint =
            cons (synthExpr typeHint) $
            emptyList
          matchingExprs typeHint et ef $ \tp t f ->
-           case asBaseType tp of
-             NotBaseType ->
-               let msg = T.concat [ "conditional where branches have base type (got "
+          case tp of
+            FloatRepr fi ->
+               return $ SomeE tp $ EApp $ FloatIte fi c t f
+            (asBaseType -> AsBaseType bty) ->
```

Comment:
I'm curious about this notation, especially in comparison to line 897.  I think I understand the notation on 897, but not this one.  Is it a pattern notation?  I don't necessarily think it's wrong, I just don't know what/how it works.

Reply:
Yeah, this is using an extension called "view patterns", and it's functionally the same as the pattern on line 897.  View patterns pass a value through a function before pattern matching on the result.

Now that you've pointed it out, I like the view pattern style better here.